### PR TITLE
feat: custom preview and live host

### LINKED ISSUE #359: Add ability to configure preview and live domain
**Is your feature request related to a problem? Please describe.**
Currently the sidekick always uses the preview domain as `*.hlx.page` and live domain as `*.hlx.live`. This does not work for situations where there's a custom domain setup for preview or live. While a separate preview or live domain isn't usually needed but can be required in certain situations like having some specific logic that needs to be put in CDN and that logic should be applicable while previewing and publishing also. For e.g. delivering some images from a different origin and not from mediabus.  

**Describe the solution you'd like**
There should be a way of configuring a preview and live domains in the sidekick. Probably it should be a "Advanced Configuration" since it would not be needed in the usual case.  The only purpose of this configs would be to use the domains specified in it while opening up the new tab for preview or live page.  Rest other functionality preview domain as `*.hlx.page` and live domain as `*.hlx.live` should remain same as usual. Maybe these could appear as separate buttons in the sidekick (like "Custom Preview", "Custom Publish") without any change in the current preview/publish buttons in what they do.

**Describe alternatives you've considered**
A workaround could be to use custom sidekick plugin to add a couple of new buttons in the sidekick which when clicked would open up the page in the configured domain.

**Additional context**
[Related conversation on this.](https://cq-dev.slack.com/archives/C9KD0TT6G/p1684142934599949?thread_ts=1683660044.408579&cid=C9KD0TT6G)

### PR DESCRIPTION
**Pull request by rofe**
Fix #359

### CODE REVIEW

**Review by trieloff — APPROVED** [review]

**Review by tripodsan — APPROVED** [review]

File: docs/API.md [hunk 1]
```diff
@@ -614,7 +614,8 @@ The sidekick configuration.
 | mountpoint | <code>string</code> |  | The content source URL (optional) |
 | project | <code>string</code> |  | The name of the project used in the sharing link (optional) |
 | plugins | [<code>Array.&lt;Plugin&gt;</code>](#Plugin) |  | An array of [plugin configurations](#Plugin) (optional) |
-| outerHost | <code>string</code> |  | The outer CDN's host name (optional) |
+| previewHost | <code>string</code> |  | The host name of a custom preview CDN (optional) |
+| liveHost | <code>string</code> |  | The host name of a custom live CDN (optional) |
```
Comment (sdmcraft) [inline docs/API.md:618]: @rofe , are these configurations available in the sidekick? If yes, where?

Comment (rofe) [reply docs/API.md:618]: @sdmcraft You can add them to the `/tools/sidekick/config.json`. I forgot to add them to the [schema](https://github.com/adobe/helix-sidekick-extension/blob/main/docs/config.schema.json), will do it now!

Comment (rofe) [reply docs/API.md:618]: https://github.com/adobe/helix-sidekick-extension/pull/411

**Pull request merged**